test/e2e/reconciler/namespace/controller_test.go

```diff
@@ -89,40 +81,27 @@ func TestNamespaceScheduler(t *testing.T) {
 				err = server.expect(namespace, unscheduledMatcher(nscontroller.NamespaceReasonUnschedulable))
 				require.NoError(t, err, "did not see namespace marked unschedulable")
 
-				t.Log("Create a cluster and expect the namespace to be scheduled to it")
-
-				server1RawConfig, err := f.Servers[physicalCluster1Name].RawConfig()
-				require.NoError(t, err, "failed to get server 1 raw config")
-
-				server1Kubeconfig, err := clientcmd.Write(server1RawConfig)
-				require.NoError(t, err, "failed to marshal server 1 kubeconfig")
+				t.Log("Creating a logical server")
+				logicalServer := framework.NewLogicalServer(t, server, server.orgClusterName)
 
-				cluster1, err := server.clusterClient.Create(ctx, &workloadv1alpha1.WorkloadCluster{
-					ObjectMeta: metav1.ObjectMeta{
-						GenerateName: "e2e-nss-1-",
-					},
-					Spec: workloadv1alpha1.WorkloadClusterSpec{
-						KubeConfig: string(server1Kubeconfig),
-					},
-				}, metav1.CreateOptions{})
+				t.Log("Create a ready cluster")
+				cluster1, err := framework.CreateReadyCluster(t, ctx, server.Artifact, server.kcpClusterClient, logicalServer)
 				require.NoError(t, err, "failed to create cluster1")
 
-				// Wait for the cluster to become ready. A namespace can only be assigned to a ready cluster.
-				waitForClusterReadiness(t, ctx, server.clusterClient, cluster1.Name)
-
 				err = server.expect(namespace, scheduledMatcher(cluster1.Name))
 				require.NoError(t, err, "did not see namespace marked scheduled for cluster1 %q", cluster1.Name)
 
 				t.Log("Cordon the cluster and expect the namespace to end up unscheduled")
 
+				wcClient := server.kcpClusterClient.WorkloadV1alpha1().WorkloadClusters()
 				err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-					cluster, err := server.clusterClient.Get(ctx, cluster1.Name, metav1.GetOptions{})
+					cluster, err := wcClient.Get(ctx, cluster1.Name, metav1.GetOptions{})
```

Comment:
always qualify clients. What's the context of wcClient? Org? Root?

Reply:
Does it make sense to be pedantic about naming when the scope is limited? The definition is 2 lines above, and it's not used anywhere else.

Reply:
Done.